src/main/java/net/minecraftforge/datafix/ForgeDataFixerDelegateHandler.java

```diff
@@ -0,0 +1,244 @@
+    }
+
+    public void rebuildFixer(final Executor executor, final boolean triggerOffThreadCompute)
+    {
+        //Rebuild the total environment.
+        //Collect all fixes.
+        final List<ForgeDataFixDelegate> globalList = new ArrayList<>(this.vanillaFixers);
+
+        //Collect all schema version keys with known fixers.
+        final IntSortedSet fixerVersions = new IntAVLTreeSet(this.vanillaFixerVersions);
+
+        //And now collect all custom fixers again that we know of and update.
+        getAllSchemas().forEach(schema -> {
+            schema.getCustomFixers().forEach(fix -> {
+                final int version = DataFixUtils.getVersion(fix.getVersionKey());
+
+                if (version > dataVersion)
+                {
+                    LOGGER.warn("Ignored fix registered for version: {} as the DataVersion of the game is: {}", version, dataVersion);
+                    return;
+                }
+
+                final ForgeDataFixDelegate delegate = new ForgeDataFixDelegate(fix);
+
+                globalList.add(delegate);
+                fixerVersions.add(fix.getVersionKey());
+            });
+          });
+
+        //Reset our rules, normal vanilla rules (ours too) keep a cache of their schemas
+        //We can not use that and need to forcefully recompute them.
+        this.vanillaFixers.forEach(ForgeDataFixDelegate::resetRule);
+
+        //All schemas are already reset via the mod event bus as this time.
+        //Rebuild of those is not needed.
+        this.activeFixer = new InnerFixer(new Int2ObjectAVLTreeMap<Schema>(schemas), new ArrayList<>(globalList), new IntAVLTreeSet(fixerVersions));
+
+        //This section was added to make it easier to debug the rule set at runtime, since it is no longer pre-computed off thread,
+        //But only on the main thread, at the moment it is needed.
+        //This basically disables the REWRITE_CACHE prefill and fills it on-demand.
+        //Although it is mainly used for debugging, it also provides a good way of reducing DFU's memory footprint, when it is not needed.
+        //Since the following code will compute an update rule from every version to the current for every known type in the schema,
+        //relating to this version. By default, however this is disabled to preserve vanilla compatibility, since it does do the pre-compute
+        //to fill up the cache.
+        //To enable it start the JVM with the system property set to true, -Dforge.datafixer.disablePreCompute=true needs to be added as
+        //a JVM Launch argument.
+        if (Boolean.getBoolean("forge.datafixer.disablePreCompute"))
+        {
+            LOGGER.warn("The pre-compute of the data fixer rules is disabled, this is not recommended for normal use, but can be useful for debugging or in low-memory situations.");
+            return;
+        }
+
+        //We only rebuild once we are told to do so.
+        if (!triggerOffThreadCompute) {
```

Comment:
Formatting: braces on newlines.
```suggestion
        if (!triggerOffThreadCompute)
        {
```